### PR TITLE
New: Added drafts in GSD / #focus mode

### LINKED ISSUE #3782: [HOLD for payment July 7] Focus mode - Show draft messages in #focus
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!

Upwork link: https://www.upwork.com/jobs/~0130bb846f8492f917
___

## Action Performed:
Scenario 1
1. Log in to e.cash
2. Navigate to a conversation and type a message (do not send it) 
3. Navigate to LHN 
4. Navigate to Preferences
5. Enable focus mode
6. Navigate to LHN 

## Expected Result:
User should be able to see drafted messages in #focus mode 

## Actual Result:
Drafted messages are not displayed in focus mode. 

## Workaround:
N/A

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS✔️ 
 Android✔️ 
 Desktop App✔️ 
 Mobile Web✔️ 

**Version Number:** 1.0.74-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:**

image

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 
From @quinthar https://expensify.slack.com/archives/C01GTK53T8Q/p1624655927132900

> When in #focus mode, draft messages should remain in the LHN even when you navigate away, even if the chat itself isn't pinned.  However, if you write a message in an unpinned chat and then navigate away before sending, the chat is removed from the LHN.

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

Added support for draft messages filter in Options for Sidebar.

### Fixed Issues

$ Fixes #3782

### Test | QA Steps

    1. Log in to e.cash
    2. Navigate to a conversation and type a message (do not send it)
    3. Navigate to LHN
    4. Navigate to Preferences
    5. Enable focus mode
    6. Navigate to LHN
    7. You should all the chats in LHn that have a draft message.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/24370807/123990485-70519680-d9e7-11eb-9192-97870748b63a.mp4

#### Mobile Web

https://user-images.githubusercontent.com/24370807/123991317-3d5bd280-d9e8-11eb-9142-c60a4b000687.mp4

#### Desktop

https://user-images.githubusercontent.com/24370807/124058657-6a8b9d80-da47-11eb-94d5-eb1ca7db0cb6.mp4

#### Android

https://user-images.githubusercontent.com/24370807/123992723-7ea0b200-d9e9-11eb-8d25-7146526e8546.mp4

### CODE REVIEW

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -296,10 +297,19 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
...
         }
 
+        const hasDraftComment = report
+            && draftComments
+            && lodashGet(draftComments, `${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${report.reportID}`, '').length > 0;
+
+
         const shouldFilterReportIfEmpty = !showReportsWithNoComments && report.lastMessageTimestamp === 0;
         const shouldFilterReportIfRead = hideReadReports && report.unreadActionCount === 0;
+        const shouldShowReportIfhasDraft = showReportsWithDrafts && hasDraftComment;
```
Comment (Jag96) [inline src/libs/OptionsListUtils.js:307]:
```suggestion
        const shouldShowReportIfHasDraft = showReportsWithDrafts && hasDraftComment;
```

**Review by Jag96 — CHANGES_REQUESTED** [review]
Looks good, 2 minor comments but this works well!

**Comment by parasharrajat** [comment]
@Jag96 Updated.

**Review by Jag96 — APPROVED** [review]
LGTM

**Comment by roryabraham** [comment]
Unfortunately the deploy comments are not working right now. This was deployed to staging yesterday.

**Pull request merged**